src/System.Private.ServiceModel/tools/setupfiles/EnsureBridgeRunning.cmd

```diff
@@ -30,9 +20,29 @@ if '%BridgeResourceFolder%' == '' (
 
 pushd %~dp0..\..\..\..\bin\wcf\tools\Bridge
 echo Invoking Bridge.exe -require -BridgeResourceFolder:%_bridgeResourceFolder% %*
-call Bridge.exe -require -BridgeResourceFolder:%_bridgeResourceFolder% %*
+call Bridge.exe -require -BridgeResourceFolder:%_bridgeResourceFolder%  %*
+set BridgeReturnCode=%ERRORLEVEL%
+
+echo Bridge.exe returned exit code %BridgeReturnCode%
+
+if '%BridgeReturnCode%' == '2' goto weStartedBridge
+if '%BridgeReturnCode%' == '1' goto bridgeAlreadyRunning
+echo EnsureBridgeRunning: The attempt to start the Bridge failed with exit code %BridgeReturnCode%
+goto done
+
+:weStartedBridge
+  echo EnsureBridgeRunning: The Bridge was started in response to this request
+  set BridgeStartedByScript=true
+  goto done
+  
+:bridgeAlreadyRunning
+  echo EnsureBridgeRunning: The Bridge was already running
+  set BridgeStartedByScript=false
+  goto done
+  
+:done
 popd
 
-
+endlocal & set BridgeStartedByScript=%BridgeStartedByScript%
 exit /b
```

Comment:
Agreed -- losing any global environment variable and relying on exit /b to propagate return code through the scripts.